fix: Fix hazmat panic

rklaehn opened the pull request:
Fixes https://github.com/n0-computer/sendme/issues/87

Unrelated changes:
- add a check to prevent people from sharing `.`. Once we have the store fully in mem this would be allowed, but not yet.
- add a `recv` alias for receive, since I keep entering `sendme recv`...

Todo:
- [ ] publish new iroh-blobs and depend on it

flub reviewed (approved):
LGTM on it's own, but I think this might be superseded by #89?
/cc @rklaehn

ramfox commented:
thanks for the review & heads up @flub !

closing because it is no longer relevant w/ #89 !

rklaehn committed:
feat: add check to prevent people from sharing from the current directory

rklaehn committed:
WIP use newest iroh-blobs

rklaehn committed:
Add an alias "recv" for receive

This is unrelated to the bugfix of course, but I have entered sendme recv
soo many times...

rklaehn committed:
Add test to make sure that attempting to send the current dir will fail

rklaehn committed:
fmt

rklaehn committed:
Add test to make sure recv fails when the send side data is modified

rklaehn commented:
Checked using a git dep that this PR https://github.com/n0-computer/iroh-blobs/pull/82 fixes this issue.

rklaehn commented:
Implemented on main.
